### ISSUE TITLE
Replicate gzip.open functionality for text files? Unable to open .gz file in text mode

### ISSUE THREAD

**Issue by jwest75674** [RaRe-Technologies/smart_open#493]
#### Existing functionality:
```
import smart_open
with smart_open.open('myfile.gz','rt') as f:
    for line in f:
        print(type(line))
        break

ValueError                                Traceback (most recent call last)
<ipython-input-8-7b490bd872da> in <module>
----> 1 with smart_open.open('myfile.gz','rt') as f:
      2     for line in f:
      3         print(type(line))
      4         break
      5

~/anaconda3/envs/test/lib/python3.7/site-packages/smart_open/smart_open_lib.py in open(uri, mode, buffering, encoding, errors, newline, closefd, opener, ignore_ext, transport_params)
    226         decompressed = binary
    227     else:
--> 228         decompressed = compression.compression_wrapper(binary, mode)
    229
    230     if 'b' not in mode or explicit_encoding is not None:

~/anaconda3/envs/test/lib/python3.7/site-packages/smart_open/compression.py in compression_wrapper(file_obj, mode)
     92         return file_obj
     93     else:
---> 94         return callback(file_obj, mode)
     95
     96

~/anaconda3/envs/test/lib/python3.7/site-packages/smart_open/compression.py in _handle_gzip(file_obj, mode)
     60 def _handle_gzip(file_obj, mode):
     61     import gzip
---> 62     return gzip.GzipFile(fileobj=file_obj, mode=mode)
     63
     64

~/anaconda3/envs/test/lib/python3.7/gzip.py in __init__(self, filename, mode, compresslevel, fileobj, mtime)
    162
    163         if mode and ('t' in mode or 'U' in mode):
--> 164             raise ValueError("Invalid mode: {!r}".format(mode))
    165         if mode and 'b' not in mode:
    166             mode += 'b'

ValueError: Invalid mode: 'rt'
```

#### Example of desired outcome:
```
with gzip.open('myfile.gz','rt') as f:
    for line in f:
        print(type(line))
        break

str
```

##
It appears that smart_open does not handle mode='rt' at the moment, requiring additional code to decode every time smart_open is used.
Requesting smart_open to match the functionality used by the standard library gzip module's open function, or help me understand the thought process as to the current setup (for my own learning purposes)

**Comment by mpenkov** [comment]
Thank you for providing a reproducible example. I've reproduced the problem and confirmed it as a bug.

Until we fix the bug, you should be able to do this instead, to achieve the desired result:

```python
import smart_open
with smart_open.open('myfile.gz','r') as f:
    for line in f:
        print(type(line))
        break
```

**Comment by mpenkov** [comment]
Fixed in db14442
